packages/cli/src/utils/build.js

```diff
@@ -332,24 +353,69 @@ const _buildFunc = async ({
     osTmpDir,
     'zapier-' + crypto.randomBytes(4).toString('hex')
   );
+  debug('Using temp directory: ', tmpDir);
 
-  maybeNotifyAboutOutdated();
+  if (checkOutdated) {
+    maybeNotifyAboutOutdated();
+  }
 
   await maybeRunBuildScript();
 
   // make sure our directories are there
   await ensureDir(tmpDir);
   await ensureDir(constants.BUILD_DIR);
 
-  startSpinner('Copying project to temp directory');
-  await copyDir(wdir, tmpDir, {
-    filter: skipNpmInstall ? (dir) => !dir.includes('.zip') : undefined,
-  });
+  if (printProgress) {
+    startSpinner('Copying project to temp directory');
+  }
+
+  const copyFilter = skipNpmInstall
+    ? (src) => !src.endsWith('.zip')
+    : undefined;
+
+  await copyDir(wdir, tmpDir, { filter: copyFilter });
+
+  if (skipNpmInstall) {
+    const corePackageDir = findCorePackageDir();
+    const nodeModulesDir = path.dirname(corePackageDir);
+    const workspaceRoot = path.dirname(nodeModulesDir);
+    if (wdir !== workspaceRoot) {
+      // If we're in here, it means the user is using npm/yarn workspaces
+      const workspaces = listWorkspaces(workspaceRoot);
+
+      await copyDir(nodeModulesDir, path.join(tmpDir, 'node_modules'), {
+        filter: (src) => {
+          if (src.endsWith('.zip')) {
+            return false;
+          }
+          const stat = fse.lstatSync(src);
+          if (stat.isSymbolicLink()) {
+            const realPath = path.resolve(
+              path.dirname(src),
+              fse.readlinkSync(src)
+            );
+            for (const workspace of workspaces) {
+              if (minimatch(realPath, workspace)) {
```

Comment:
@eliangcs Could we add a comment here explaining what this `minimatch` line is doing? I believe it's `true` if the `src` symlink resolves (points to) to a workspace path, in which case we want to filter out (`return false`) this particular `src`. So that we don't copy over any symlinks that point to workspace folders.

Reply:
You interpretation is correct. Since the `workspaces` array in the root `package.json` can contain glob patterns, like `packages/*`, I use minimatch to help us do glob pattern matching. I'll add a comment in a separate PR. Thanks!